### PR TITLE
some bug fixes for todos 

### PR DESCRIPTION
**Pull request by gtalha07**
- Fixes for #457.
- Apply validation for leading and trailing whitespaces in todos creation input.

### CODE REVIEW

File: app/lib/screens/HomeScreens/todo/screens/CreateTodo.dart [hunk 1]
```diff
@@ -147,7 +147,7 @@ class _CreateTodoScreenState extends State<CreateTodoScreen> {
           controller: descriptionController,
           keyboardType: TextInputType.text,
           inputFormatters: <TextInputFormatter>[
-            FilteringTextInputFormatter.allow(RegExp(r'[0-9a-zA-Z]+|\s')),
+            FilteringTextInputFormatter.deny(RegExp(r'^[^a-z0-9]$')),
```
Comment (gnunicorn) [inline app/lib/screens/HomeScreens/todo/screens/CreateTodo.dart:150]: same here. why limit the input?

**Comment by gnunicorn** [comment]
@JMEFK noticed earlier today that no non ascii is allowed in the text fields. I think this is related to the regexp you have in there. Why is there a regex?

**Review by bitfriend — APPROVED** [review]
Looks good to me

File: app/lib/controllers/todo_controller.dart [hunk 2]
```diff
@@ -248,14 +248,14 @@ class ToDoController extends GetxController {
   }
 
   //ToDo list card expand.
-  void toggleCardExpand(int index) {
-    cardExpand = !cardExpand;
+  void toggleCardExpand(int index, bool prevState) {
```
[the remark this answers is not in the document]
Comment (gtalha07) [reply app/lib/controllers/todo_controller.dart:252]: We don't have any explicit ```id``` or unique identifier for the todos itself. The ```index``` serves as a position where the todo is placed in ```ListView``` which of course will gets updated whenever list length changes :)

[the remark this answers is not in the document]
Comment (gtalha07) [reply app/lib/controllers/todo_controller.dart:252]: The index comes as prop from the child position of ```ListView```.

**Pull request closed without merging**